lib/seahorn/BvOpSem2RawMemMgr.cc

```diff
@@ -817,5 +817,9 @@ bool RawMemManagerCore::isPtrTyVal(Expr e) {
   return (!e || !strct::isStructVal(e));
 }
 
+bool RawMemManagerCore::isMemVal(Expr e) {
+  return (!e || !strct::isStructVal(e));
```

Comment:
Are we ok accepting a `NULL` value for e? If so, why?

I would assume it should be `e && !strct::isStructVal(e)`

Note when constructing a `MemValTy`, we allow `NULL` values since that is a mechanism to signal that the resulting `MemValTy` is bad (e.g. an operation failed). I don't see a use case where we take a bad `MemValTy` and then `getRaw()` on that.

Reply:
good point, when I first wrote this I was just trying to follow the constructor of `MemValTy`; Let's make all `isMemVal` return false when `e` is NULL and in that case `getRawMem` will just return an empty `Expr` (NULL); should probably do the same for `isPtrTyVal`